tools/please_pex/pex/pex_main.py

```diff
@@ -175,18 +168,22 @@ def __init__(self, module_dir=MODULE_DIR):
 
     def _find_all_distributions(self, module_dir):
         dists = {}
-        if zipfile.is_zipfile(sys.argv[0]):
-            zf = ZipFileWithPermissions(sys.argv[0])
+        pex_file = sys.argv[0]
+        if zipfile.is_zipfile(pex_file):
```

Comment:
Sidebar since it was already here: is there any world in which the input is not a zipfile? Should we either log/raise in an else clause, or just remove the `if` and let an exception happen?

Reply:
I _think_ (although I'm not sure, because this isn't really documented very well upstream) that some `MetaPathFinder` abstract methods are supposed to return `None` if the object can't successfully answer a given query. We've had bad experiences recently when raising exceptions in this code. If we put a log line in here, I suspect it'll be spamming stderr constantly.

Reply:
Yes, that matches my understanding; MetaPathFinders don't have to handle a particular request. I agree that logging would be too noisy.

I think it might turn out not to be a zipfile if the pex was marked zip-unsafe; or maybe there is some other edge case (e.g. what if people get adventurous with `multiprocessing`, I've seen some weird outcomes there)